src/psyclone/tests/psyir/nodes/dynamic_omp_task_directive_test.py

```diff
@@ -0,0 +1,3915 @@
+      type :: y
+        integer, dimension(321) :: jp
+      end type
+      type :: x
+        type(y), dimension(5) :: y
+      end type
+        integer :: i, ii
+        integer :: j, jj
+        integer :: k
+        type(x) :: ty
+
+        do i = 1, 320, 32
+            do j = 1, 320, 32
+                do ii=i, i+32
+                    do jj = j,j+32
+                        k = ty%y%jp(ii) + ii
+                        ty%y(2)%jp(ii+1) = ty%y(1)%jp(ii+1) - (1 - \
+ty%y(3)%jp(ii+1))
+                    end do
+                end do
+            end do
+        end do
+    end subroutine
+    '''
+    tree = fortran_reader.psyir_from_source(code)
+    ptrans = OMPParallelTrans()
+    strans = OMPSingleTrans()
+    tdir = DynamicOMPTaskDirective()
+    loops = tree.walk(Loop)
+    loop = loops[0].children[3].children[0]
+    parent = loop.parent
+    loop.detach()
+    tdir.children[0].addchild(loop)
+    parent.addchild(tdir, index=0)
+    strans.apply(tree.children[0].children[:])
+    ptrans.apply(tree.children[0].children[:])
+    with pytest.raises(GenerationError) as excinfo:
+        tree.lower_to_language_level()
+    assert("PSyclone doesn't support an OMPTaskDirective containing a "
+           "StructureReference with multiple "
+           "array accessing members. Found 'ty%y(2)%jp(ii + 1)'"
+           in str(excinfo.value))
+
+
+def test_omp_task_directive_write_index_shared_type(fortran_reader):
+    ''' Test the code generation correctly generates an error if an array index
+    of a written array is a shared variable.'''
+    code = '''
+    subroutine my_subroutine()
+        type :: x
+          integer, dimension(320, 10) :: A
+        end type
+        type(x) :: AA
+        integer, dimension(321, 10) :: B
+        integer :: i
+        integer :: j
+        integer :: k
+        do i = 1, 320, 32
+            k = 4
+            k = -2
+            k = k + 3
+            do j = 1, 32
+                AA%A(i, k) = k
+                AA%A(i, j) = B(1+i, j) + k
+            end do
+        end do
+    end subroutine
+    '''
+    tree = fortran_reader.psyir_from_source(code)
+    ptrans = OMPParallelTrans()
+    strans = OMPSingleTrans()
+    tdir = DynamicOMPTaskDirective()
+    loops = tree.walk(Loop, stop_type=Loop)
+    loop = loops[0].children[3].children[3]
+    parent = loop.parent
+    loop.detach()
+    tdir.children[0].addchild(loop)
+    parent.addchild(tdir, index=0)
+    strans.apply(loops[0])
+    ptrans.apply(loops[0].parent.parent)
+
+    with pytest.raises(GenerationError) as excinfo:
+        tree.lower_to_language_level()
+    assert ("Shared variable access used as an array index inside an "
+            "OMPTaskDirective which is not supported. Variable name is "
+            "'Reference[name:'k']'" in str(excinfo.value))
+
+
+def test_omp_task_directive_read_index_shared_type(fortran_reader):
+    ''' Test the code generation correctly generates an error if an array index
```

Comment:
Please extend. Presumably something about a structure member to differentiate it from previous tests?

Reply:
Yeah - added the structure mention.